### PR TITLE
Added listTopics() and describeTopics() method in KafkaAdminClient.

### PR DESCRIPTION
**Pull request by s-ourabh**

### CODE REVIEW

File: clients/src/main/java/org/oracle/okafka/clients/admin/KafkaAdminClient.java [hunk 1]
```diff
@@ -129,12 +130,16 @@
...
+import org.apache.kafka.clients.admin.TopicDescription;
+import org.apache.kafka.clients.admin.TopicListing;
 import org.apache.kafka.clients.admin.UnregisterBrokerOptions;
 import org.apache.kafka.clients.admin.UnregisterBrokerResult;
 import org.apache.kafka.clients.admin.UpdateFeaturesOptions;
 import org.apache.kafka.clients.admin.UpdateFeaturesResult;
 import org.apache.kafka.clients.admin.UserScramCredentialAlteration;
 import org.apache.kafka.clients.admin.CreateTopicsResult.TopicMetadataAndConfig;
+//import org.apache.kafka.clients.admin.KafkaAdminClient.LeastLoadedNodeProvider;
+//import org.apache.kafka.clients.admin.KafkaAdminClient.LeastLoadedNodeProvider;
```
Comment (ichokshi2109) [inline clients/src/main/java/org/oracle/okafka/clients/admin/KafkaAdminClient.java:142]: delete these

File: clients/src/main/java/org/oracle/okafka/clients/admin/KafkaAdminClient.java [hunk 2]
```diff
@@ -1680,32 +1700,24 @@ void handleFailure(Throwable throwable) {
...
-				new org.oracle.okafka.clients.admin.DeleteTopicsOptions());
-		KafkaFuture<Void> ftr = delResult.all();
-		try {
-			ftr.get();
-		} catch (Exception e) {
-			log.error("Exception from deleteTopics", e);
+	public DeleteTopicsResult deleteTopics(final TopicCollection topics, DeleteTopicsOptions options) {
+
+		if (topics instanceof TopicIdCollection) {
+			throw new FeatureNotSupportedException("Topic Id's are not supported for deleteTopics request");
```
Comment (ichokshi2109) [inline clients/src/main/java/org/oracle/okafka/clients/admin/KafkaAdminClient.java:1706]: Topic Ids are  not supported for this release of OKafka.

Comment (s-ourabh) [reply clients/src/main/java/org/oracle/okafka/clients/admin/KafkaAdminClient.java:1706]: Okay.

Comment (s-ourabh) [reply clients/src/main/java/org/oracle/okafka/clients/admin/KafkaAdminClient.java]: I changed it in this same pull request.

File: clients/src/main/java/org/oracle/okafka/common/network/AQClient.java [hunk 3]
```diff
@@ -117,82 +117,158 @@ public AQClient(Logger log, AbstractConfig configs) {
...
+
+			}
+
+			}catch (Exception exception) {
+				log.error("Exception while listing topics " + exception.getMessage(), exception);
+
+				if (exception instanceof SQLException)
+					if (((SQLException) exception).getErrorCode() == 6550) {
+						log.error("Not all privileges granted to the database user.",
+								((SQLException) exception).getMessage());
```
Comment (ichokshi2109) [inline clients/src/main/java/org/oracle/okafka/common/network/AQClient.java:222]: log.error(message, exception) . We should have a complete information.

File: clients/src/main/java/org/oracle/okafka/common/network/AQClient.java [hunk 4]
```diff
@@ -117,82 +117,158 @@ public AQClient(Logger log, AbstractConfig configs) {
...
+						log.info("Please grant all the documented privileges to database user.");
```
Comment (ichokshi2109) [inline clients/src/main/java/org/oracle/okafka/common/network/AQClient.java:223]: You can remove this line.

File: clients/src/test/java/org/oracle/okafka/tests/OkafkaDescribeTopics.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+        	}
+        	
+		}
+		catch(Exception e)
+		{
+			System.out.println("Exception while creating topic " + e);
+			e.printStackTrace();
+		}
+		
+		System.out.println("Main thread complete ");
```
Comment (ichokshi2109) [inline clients/src/test/java/org/oracle/okafka/tests/OkafkaDescribeTopics.java:41]: Change it to say Test :DescribeTopic complete.

File: clients/src/test/java/org/oracle/okafka/tests/OkafkaDescribeTopics.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+        	Map<String,KafkaFuture<TopicDescription>> description=res.topicNameValues();
+        	for(Map.Entry<String,KafkaFuture<TopicDescription>> entry : description.entrySet()) {
+        		System.out.println("Description - "+entry.getValue().get());
+        	}
+        	
+		}
+		catch(Exception e)
+		{
+			System.out.println("Exception while creating topic " + e);
```
Comment (ichokshi2109) [inline clients/src/test/java/org/oracle/okafka/tests/OkafkaDescribeTopics.java:37]: Wrong exception message.

File: clients/src/test/java/org/oracle/okafka/tests/OkafkaListTopics.java [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+import org.apache.kafka.clients.admin.ListTopicsOptions;
+import org.apache.kafka.clients.admin.ListTopicsResult;
+
+public class OkafkaListTopics {
+
+	@Test
+	public void AdminTest() {
+        try (Admin admin = AdminClient.create(OkafkaSetup.setup())) {
+        	
+        	KafkaAdminClient kAdminClient = (((org.oracle.okafka.clients.admin.KafkaAdminClient)admin));
```
Comment (ichokshi2109) [inline clients/src/test/java/org/oracle/okafka/tests/OkafkaListTopics.java:17]: why do we need to typecaste?

File: clients/src/test/java/org/oracle/okafka/tests/OkafkaDescribeTopics.java [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+
+public class OkafkaDescribeTopics {
+
+	@Test
+	public void AdminTest() {
+        try (Admin admin = AdminClient.create(OkafkaSetup.setup())) {
+        	
+
+
+        	KafkaAdminClient kAdminClient = (((org.oracle.okafka.clients.admin.KafkaAdminClient)admin));
```
Comment (ichokshi2109) [inline clients/src/test/java/org/oracle/okafka/tests/OkafkaDescribeTopics.java:25]: Why do we need to typecaste?

**Pull request closed without merging**